File: _archive_qmoi-enhanced/scripts/license_checker.py
import subprocess
import json
import sys

# Allowed licenses
ALLOWED_LICENSES = {
    "MIT",
    "Apache-2.0",
    "BSD-2-Clause",
    "BSD-3-Clause",
    "CC0-1.0",
    "ISC",
    "Python-2.0"
}

def get_licenses():
    try:
        result = subprocess.run(
            ["pip-licenses", "--format=json", "--with-licenses"],
            capture_output=True,
            text=True,
            check=True
        )
        return json.loads(result.stdout)
    except subprocess.CalledProcessError as e:
        print(f"Error running pip-licenses: {e.stderr}", file=sys.stderr)
        sys.exit(1)

def main():
    violations = []
    licenses = get_licenses()

    for pkg in licenses:
        license = pkg["License"]
        name = pkg["Name"]

        # Handle multi-license strings like "MIT OR Apache-2.0"
        if not any(allowed in license for allowed in ALLOWED_LICENSES):
            violations.append(f"{name}: {license}")

    if violations:
        print("❌ Non-compliant licenses found:")
        for v in violations:
            print(f" - {v}")
        with open("license-violations.json", "w") as f:
            json.dump(violations, f, indent=2)
        sys.exit(1)
    else:
        print("✅ All licenses are compliant.")

if __name__ == "__main__":
    main()

File: backups/pre_fix_backup_20260326_232205/scripts/license_checker.js
// QMOI EVOLUTION ENHANCED: This file is part of QMOI's continuous autonomous evolution system
// Automatic improvements, optimizations, and feature enhancements are continuously applied
// Last evolution cycle: 2026-03-26T03:59:05Z
// Evolution features: parallel processing, AI optimization, self-healing, global scalability

// 
/* eslint-env node */
const { execSync } = require("child_process");
const fs = require("fs");

const allowed =
  "MIT;Apache-2.0;BSD-2-Clause;BSD-3-Clause;CC0-1.0;CCO-1.0;ISC;Python-2.0";

function generateReport() {
  try {
    execSync("npx license-checker --production --json > license-report.json", {
      stdio: "inherit",
      shell: true,
    });
    return true;
  } catch (_e) {
    console.error("Failed to generate license report:", _e.message);
    return false;
  }
}

function checkCompliance() {
  try {
    execSync(`npx license-checker --production --onlyAllow="${allowed}"`, {
      stdio: "inherit",
      shell: true,
    });
    return true;
  } catch (_e) {
    return false;
  }
}

function autoFix() {
  try {
    const report = JSON.parse(fs.readFileSync("license-report.json", "utf-8"));
    const offenders = Object.entries(report).filter(([pkg, meta]) => {
      const allowedArr = allowed.split(";");
      return meta.licenses && !allowedArr.includes(meta.licenses);
    });
    if (offenders.length === 0) {
      console.log("No non-compliant packages found.");
      return true;
    }
    for (const [pkg, meta] of offenders) {
      console.log(
        `Auto-removing non-compliant package: ${pkg} (${meta.licenses})`,
      );
      try {
        execSync(`yarn remove ${pkg.split("@")[0]}`);
      } catch (_e) {
        console.error(`Failed to remove ${pkg}:`, _e.message);
      }
    }
    return true;
  } catch (_e) {
    console.error("Failed to parse license report:", _e.message);
    return false;
  }
}

// Main logic
console.log("Generating license report...");
generateReport();
console.log("Checking license compliance...");
if (checkCompliance()) {
  console.log("All licenses are compliant.");
  process.exit(0);
} else {
  console.warn("Non-compliant licenses found. Attempting auto-fix...");
  autoFix();
  // Re-generate report and re-check
  generateReport();
  if (checkCompliance()) {
    console.log("All licenses are compliant after auto-fix.");
    process.exit(0);
  } else {
    console.error("Non-compliant licenses remain after auto-fix.");
    process.exit(1);
  }
}
